runtime/runtime-params-estimator/src/action_costs.rs

```diff
@@ -0,0 +1,616 @@
+        .inner_iters(1) // only one account deletion possible
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn delete_account_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(delete_account_action())
+        .inner_iters(1) // only one account deletion possible
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn delete_account_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(delete_account_action())
+        .inner_iters(1) // only one account deletion possible
+        .apply_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn deploy_contract_base_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(deploy_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn deploy_contract_base_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(deploy_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+/// Note: This is not the best estimation because a dummy contract is clearly
+/// not the worst-case scenario for gas costs.
+pub(crate) fn deploy_contract_base_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(deploy_action(ActionSize::Min))
+        .apply_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn deploy_contract_byte_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(deploy_action(ActionSize::Max))
+        .inner_iters(1) // circumvent TX size limit
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.deploy_contract()
+}
+
+pub(crate) fn deploy_contract_byte_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(deploy_action(ActionSize::Max))
+        .inner_iters(1) // circumvent TX size limit
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.deploy_contract()
+}
+
+pub(crate) fn deploy_contract_byte_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(deploy_action(ActionSize::Max))
+        .inner_iters(1) // circumvent TX size limit
+        .apply_cost(&mut ctx.testbed())
+        / ActionSize::Max.deploy_contract()
+}
+
+pub(crate) fn function_call_base_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new_sir(ctx)
+        .add_action(function_call_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn function_call_base_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn function_call_base_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
```

Comment:
In some cases the `exec` estimation uses a SiR action, and in others it uses the non-SiR one. Any reason for the discrepancy (AFAIU all methods that are non-SiR compatible, are also SiR compatible, right?) I’d say either make it consistent everywhere or document what the preference is and why pick one over the other.

Reply:
Yes, you are right, we can use `new_sir` for all `exec` estimations. It really shouldn't matter for the execution, so let's just be consistent.

Thanks for the great review on this PR, it was very helpful!